### PR TITLE
feat: Update to copy from disruption summary link

### PR DESCRIPTION
**Pull request by alainajaffer**
Also added a request from the figma design ticket

### CODE REVIEW

File: site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx [hunk 1]
```diff
@@ -113,7 +113,7 @@ const CreateConsequenceNetwork = (props: CreateConsequenceNetworkProps): ReactEl
                                     props.disruptionSummary ? stateUpdater(props.disruptionSummary, "description") : "";
                                 }}
                             >
-                                <p className="text-govBlue govuk-body-m">Copy from disruption summary</p>
+                                <p className="text-govBlue govuk-body-m">Copy from disruption description</p>
```
[the remark this answers is not in the document]
Comment (ljones92) [reply site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx:116]: @alvak1 I've actually fixed that issue in my twitter api branch

[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx:116]: fixed on my branch too

File: site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx [hunk 2]
```diff
@@ -113,7 +113,7 @@ const CreateConsequenceNetwork = (props: CreateConsequenceNetworkProps): ReactEl
                                     props.disruptionSummary ? stateUpdater(props.disruptionSummary, "description") : "";
```
Comment (DeannaSharma) [inline site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx:113]: Can we change the name of this prop to disruptionDescription to reflect it is copy the description not a summary

File: site/pages/create-consequence-operator/[disruptionId]/[consequenceIndex].page.tsx [hunk 3]
```diff
@@ -209,7 +209,7 @@ const CreateConsequenceOperator = (props: CreateConsequenceOperatorProps): React
                                     props.disruptionSummary ? stateUpdater(props.disruptionSummary, "description") : "";
```
Comment (DeannaSharma) [inline site/pages/create-consequence-operator/[disruptionId]/[consequenceIndex].page.tsx:209]: Can we change the name of this prop to disruptionDescription to reflect it is copy the description not a summary

File: site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx [hunk 4]
```diff
@@ -508,7 +508,7 @@ const CreateConsequenceServices = (props: CreateConsequenceServicesProps): React
                                     props.disruptionSummary ? stateUpdater(props.disruptionSummary, "description") : "";
```
Comment (DeannaSharma) [inline site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx:508]: Can we change the name of this prop to disruptionDescription to reflect it is copy the description not a summary

File: site/pages/create-consequence-stops/[disruptionId]/[consequenceIndex].page.tsx [hunk 5]
```diff
@@ -291,7 +291,7 @@ const CreateConsequenceStops = (props: CreateConsequenceStopsProps): ReactElemen
                                     props.disruptionSummary ? stateUpdater(props.disruptionSummary, "description") : "";
```
Comment (DeannaSharma) [inline site/pages/create-consequence-stops/[disruptionId]/[consequenceIndex].page.tsx:291]: Can we change the name of this prop to disruptionDescription to reflect it is copy the description not a summary

**Review by alvak1 — DISMISSED** [review]
Approved

File: site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx [hunk 6]
```diff
@@ -92,7 +92,7 @@ const CreateSocialMediaPost = (props: CreateSocialMediaPostPageProps): ReactElem
                                     stateUpdater(props.disruptionSummary, "messageContent");
```
Comment (DeannaSharma) [inline site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx:92]: For social media posts, we are copying over the summary not the description - is this intended behaviour as ticket says to copy over disruption description?

If we do need to copy over the disruption description then we may face issues with character length. A disruption description can be 1000 characters whereas a social media post can be 200 characters.

Can we get some clarification from Chris?

Comment (alainajaffer) [reply site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx:92]: he said to increase

File: site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx (hunk 4 above)
Comment (alainajaffer) [reply site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx]: done

File: site/pages/create-consequence-stops/[disruptionId]/[consequenceIndex].page.tsx (hunk 5 above)
Comment (alainajaffer) [reply site/pages/create-consequence-stops/[disruptionId]/[consequenceIndex].page.tsx]: done

File: site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx (hunk 6 above)
Comment (alainajaffer) [reply site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx]: done

File: site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx (hunk 2 above)
Comment (alainajaffer) [reply site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx]: done

File: site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx (hunk 6 above)
Comment (DeannaSharma) [reply site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx]: Social media posts are still copying the disruption summary not the disruption description.

Also I'm not 100% we can increase the limit to 1000 characters for social media posts. Twitter's character limit is 280 - I'm not sure what happens if we exceed this limit.

File: site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx [hunk 7]
```diff
@@ -240,7 +240,7 @@ export const getServerSideProps = async (ctx: NextPageContext): Promise<{ props:
     return {
         props: {
             ...getPageState(errorCookie, socialMediaPostSchema, disruptionId, socialMediaPost || undefined),
-            disruptionSummary: disruption?.summary || "",
+            disruptionDescription: disruption?.summary || "",
```
Comment (DeannaSharma) [inline site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx:243]: This needs changing to disruption?.description to copy the description

Comment (alainajaffer) [reply site/pages/create-social-media-post/[disruptionId]/[socialMediaPostIndex].page.tsx]: done

**Review by DeannaSharma — APPROVED** [review]

**Pull request merged**